scrooge-generator: add scala language-flag "immutable-sequences"

basert opened the pull request:
Problem

Scala 2.13 changed the alias for `scala.Seq[A]` from 
`scala.collection.Seq[A]` to `scala.collection.immutable.Seq[A]`.
Code generated by scrooge expects `scala.collection.Seq` in handlers
and as result from service calls.

Solution

Add a language-flag that allows explicit `scala.collection.immutable.Seq`.

Result

This keeps backwards compatibility and allows toggling the new behavior via
language-flag during code generation.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/twitter/scrooge?pullRequest=350) <br/>All committers have signed the CLA.

Commit:
scrooge-generator: add scala language-flag "immutable-sequences"

Problem

Scala 2.13 changed the alias for `scala.Seq[A]` from
`scala.collection.Seq[A]` to `scala.collection.immutable.Seq[A]`.
Code generated by scrooge expects `scala.collection.Seq` in handlers
and as result from service calls.

Solution

Add a language-flag that allows explicit `scala.collection.immutable.Seq`.

Result

This keeps backwards compatibility and allows toggling the new behavior via
language-flag during code generation.

basert commented:
I rebased to the latest develop branch and adjusted the goldfile changes. This should fix the pipelines.

jyanJing commented:
Thank you for the contribution, your change looks great to me, I have pulled your change and getting internal reviews.

jyanJing commented:
Hi @basert , your branch is [landed](https://github.com/twitter/scrooge/commit/6c290f2c638ef9971c22fe0a4f70477570f86fa1), thank you again for the contribution!
